### PR TITLE
Allow blocks to pick data sent to the adminUI via extendAdminMeta()

### PR DESCRIPTION
**Pull request by jesstelford**
Note: This is a competing implementation to #1097 based on the discussion there.

### CODE REVIEW

File: packages/fields/src/types/Content/Implementation.js [hunk 1]
```diff
@@ -227,6 +227,21 @@ export class Content extends Relationship {
...
+        .filter(([, blockConfig]) => blockConfig && Object.keys(blockConfig).length)
+        .reduce(
+          (options, block) => ({
+            ...options,
+            [block[0].type]: block[1],
+          }),
+          {}
+        ),
+
       blockOptions: this.blocks
```
Comment (mitchellhamilton) [inline packages/fields/src/types/Content/Implementation.js]: This should be removed right? Otherwise the blockOptions above would have no effect?

Comment (jesstelford) [reply packages/fields/src/types/Content/Implementation.js]: 🤦‍♂ Copy pasta error. Thanks!

File: packages/fields/src/types/Content/Implementation.js [hunk 2]
```diff
@@ -226,9 +227,22 @@ export class Content extends Relationship {
...
       blockTypes: this.blocks.map(block => (Array.isArray(block) ? block[0] : block).type),
 
-      // Key the block options by type to be serialised and passed to the client
       blockOptions: this.blocks
-        .filter(block => Array.isArray(block) && !!block[1])
+        // Normalise the input config so everything is an array
+        .map(block => (Array.isArray(block) ? [block[0], block[1]] : [block, {}]))
+        // Give complex blocks the opportunity to set default values
+        .map(([block, blockConfig]) => {
+          const complexBlockInstance = this.complexBlocks.find(({ type }) => type === block.type);
```
Comment (mitchellhamilton) [inline packages/fields/src/types/Content/Implementation.js]: Couldn't you do `this.complexBlocks.find((block) => block.constructor === block.implementation)`? and then you wouldn't need to set `this.type`?

Comment (jesstelford) [reply packages/fields/src/types/Content/Implementation.js]: Yeah, that's one way to do it. I realised I wanted to move `this.type` to the super class anyway as it's important information.

I think both implementations would work.

**Pull request merged**